influxdb3/tests/auth.rs

```diff
@@ -0,0 +1,181 @@
+use reqwest::StatusCode;
+use std::env;
+use std::process::Command;
+use std::process::Stdio;
+
+#[tokio::test]
+async fn auth() {
+    // The binary is made before testing so we have access to it
+    let bin_path = {
+        let mut bin_path = env::current_exe().unwrap();
+        bin_path.pop();
+        bin_path.pop();
+        bin_path.join("influxdb3")
+    };
+    let mut server = Command::new(bin_path)
+        .args([
+            "serve",
+            "--bearer-token",
+            "2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae", // foo as a sha256
+        ])
+        .stderr(Stdio::null())
+        .stdout(Stdio::null())
+        .spawn()
+        .expect("Was able to spawn a server");
+    let client = reqwest::Client::new();
+
+    // Wait for the server to come up
+    while client
+        .get("http://127.0.0.1:8181/health")
+        .bearer_auth("foo")
+        .send()
+        .await
+        .is_err()
+    {}
+
+    match client
+        .post("http://127.0.0.1:8181/api/v3/write_lp?db=foo")
+        .body("cpu,host=a val=1i 123")
+        .send()
+        .await
+    {
+        Err(e) => {
+            server.kill().unwrap();
+            panic!("request errored out: {e}");
+        }
+        Ok(resp) => {
+            if resp.status() != StatusCode::UNAUTHORIZED {
+                server.kill().unwrap();
+                panic!("status code was not unauthorized: {:#?}", resp);
+            }
+        }
+    }
```

Comment:
Yeah. The other problem is that if panic=abort it'll just skip this and it'll cause the program to remain open. I don't think we are and so it could be condensed, but I opted for the more verbose one in this case. I can experiment with a Drop wrapper to see if it works though!